<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>HMP_2012_WMS_gingiva - Supragingival vs subgingival plaque â€¢ MicrobiomeBenchmarkDataAnalyses</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="HMP_2012_WMS_gingiva - Supragingival vs subgingival plaque">
<meta property="og:description" content="MicrobiomeBenchmarkDataAnalyses">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MicrobiomeBenchmarkDataAnalyses</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MicrobiomeBenchmarkDataAnalyses.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/HMP_2012_16S_gingival_V13.html"> HMP_2012_16S_gingival_V13 - subgingival vs supgragingival plaque</a>
    </li>
    <li>
      <a href="../articles/HMP_2012_16S_gingival_V35_subset.html">HMP_2012_16S_gingival_V35_subset - subgingival vs supragingival</a>
    </li>
    <li>
      <a href="../articles/HMP_2012_16S_gingival_V35.html"> HMP_2012_16S_gingival_V35 - subgingival vs supgragingival plaque</a>
    </li>
    <li>
      <a href="../articles/HMP_2012_WMS_gingival.html">HMP_2012_WMS_gingiva - Supragingival vs subgingival plaque</a>
    </li>
    <li>
      <a href="../articles/Ravel_2011_16S_BV.html">Ravel_2011_16S_BV - Bacterial vaginosis</a>
    </li>
    <li>
      <a href="../articles/Stammler_2016_16S_spikein.html">Sammler_2016_16S_spikein - Spike-in bacteria</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>HMP_2012_WMS_gingiva - Supragingival vs
subgingival plaque</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses/blob/HEAD/vignettes/articles/HMP_2012_WMS_gingival.Rmd" class="external-link"><code>vignettes/articles/HMP_2012_WMS_gingival.Rmd</code></a></small>
      <div class="hidden name"><code>HMP_2012_WMS_gingival.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses" class="external-link">MicrobiomeBenchmarkDataAnalyses</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/MicrobiomeBenchmarkData" class="external-link">MicrobiomeBenchmarkData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">benchdamic</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="import-data">Import data<a class="anchor" aria-label="anchor" href="#import-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_name</span> <span class="op">&lt;-</span><span class="st">'HMP_2012_WMS_gingival'</span></span>
<span><span class="va">conditions_col</span> <span class="op">&lt;-</span> <span class="st">'body_subsite'</span></span>
<span><span class="va">conditions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>condB <span class="op">=</span> <span class="st">'subgingival_plaque'</span>, condA <span class="op">=</span> <span class="st">'supragingival_plaque'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MicrobiomeBenchmarkData/reference/getBenchmarkData.html" class="external-link">getBenchmarkData</a></span><span class="op">(</span><span class="va">dat_name</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterTaxa.html">filterTaxa</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="va">conditions_col</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="va">conditions_col</span><span class="op">]</span><span class="op">]</span>, levels <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span></span>
<span><span class="va">tse</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 189 16 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(189): Streptococcus_parasanguinis Veillonella_dispar ...</span></span>
<span><span class="co">##   Prevotella_baroniae Propionibacterium_acidifaciens</span></span>
<span><span class="co">## rowData names(8): kingdom phylum ... species taxon_annotation</span></span>
<span><span class="co">## colnames(16): SRS013949 SRS013950 ... SRS063215 SRS065310</span></span>
<span><span class="co">## colData names(24): dataset subject_id ... curator BMI</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (189 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="prior-knowledge">Prior knowledge<a class="anchor" aria-label="anchor" href="#prior-knowledge"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">row_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">prior_info</span> <span class="op">&lt;-</span> <span class="va">row_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'genus'</span>, <span class="st">'taxon_annotation'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">prior_info</span><span class="op">$</span><span class="va">taxon_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">)</span></span>
<span><span class="va">prior_info</span><span class="op">$</span><span class="va">new_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">$</span><span class="va">taxon_name</span>, <span class="st">'|'</span>, <span class="va">prior_info</span><span class="op">$</span><span class="va">genus</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                     genus      taxon_annotation</span></span>
<span><span class="co">## Streptococcus_parasanguinis Streptococcus facultative_anaerobic</span></span>
<span><span class="co">## Veillonella_dispar            Veillonella             anaerobic</span></span>
<span><span class="co">## Veillonella_atypica           Veillonella             anaerobic</span></span>
<span><span class="co">## Veillonella_parvula           Veillonella             anaerobic</span></span>
<span><span class="co">## Haemophilus_parainfluenzae    Haemophilus facultative_anaerobic</span></span>
<span><span class="co">## Dialister_invisus               Dialister             anaerobic</span></span>
<span><span class="co">##                                              taxon_name</span></span>
<span><span class="co">## Streptococcus_parasanguinis Streptococcus_parasanguinis</span></span>
<span><span class="co">## Veillonella_dispar                   Veillonella_dispar</span></span>
<span><span class="co">## Veillonella_atypica                 Veillonella_atypica</span></span>
<span><span class="co">## Veillonella_parvula                 Veillonella_parvula</span></span>
<span><span class="co">## Haemophilus_parainfluenzae   Haemophilus_parainfluenzae</span></span>
<span><span class="co">## Dialister_invisus                     Dialister_invisus</span></span>
<span><span class="co">##                                                             new_names</span></span>
<span><span class="co">## Streptococcus_parasanguinis Streptococcus_parasanguinis|Streptococcus</span></span>
<span><span class="co">## Veillonella_dispar                     Veillonella_dispar|Veillonella</span></span>
<span><span class="co">## Veillonella_atypica                   Veillonella_atypica|Veillonella</span></span>
<span><span class="co">## Veillonella_parvula                   Veillonella_parvula|Veillonella</span></span>
<span><span class="co">## Haemophilus_parainfluenzae     Haemophilus_parainfluenzae|Haemophilus</span></span>
<span><span class="co">## Dialister_invisus                         Dialister_invisus|Dialister</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="convert-to-phyloseq">Convert to phyloseq<a class="anchor" aria-label="anchor" href="#convert-to-phyloseq"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/makePhyloseqFromTreeSE.html" class="external-link">makePhyloseqFromTreeSummarizedExperiment</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">body_subsite</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">body_subsite</span><span class="op">)</span></span>
<span><span class="va">ps</span></span></code></pre></div>
<pre><code><span><span class="co">## phyloseq-class experiment-level object</span></span>
<span><span class="co">## otu_table()   OTU Table:         [ 189 taxa and 16 samples ]</span></span>
<span><span class="co">## sample_data() Sample Data:       [ 16 samples by 24 sample variables ]</span></span>
<span><span class="co">## tax_table()   Taxonomy Table:    [ 189 taxa by 7 taxonomic ranks ]</span></span>
<span><span class="co">## phy_tree()    Phylogenetic Tree: [ 189 tips and 188 internal nodes ]</span></span></code></pre>
<div class="section level3">
<h3 id="differential-abundance-analysis">Differential abundance analysis<a class="anchor" aria-label="anchor" href="#differential-abundance-analysis"></a>
</h3>
<p>Select methods for DA:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/runNormalizations.html" class="external-link">runNormalizations</a></span><span class="op">(</span><span class="fu"><a href="../reference/set_norm_list.html">set_norm_list</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">ps</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">zw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/weights_ZINB.html" class="external-link">weights_ZINB</a></span><span class="op">(</span><span class="va">ps</span>, design <span class="op">=</span> <span class="va">conditions_col</span><span class="op">)</span></span>
<span><span class="va">DA_methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/set_DA_methods_list.html">set_DA_methods_list</a></span><span class="op">(</span><span class="va">conditions_col</span>, <span class="va">conditions</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Seurat"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">contrast</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>        <span class="kw">next</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># These methods throw an error, so they must be removed</span></span>
<span><span class="co"># DA_methods &lt;- DA_methods[!names(DA_methods) == 'DA_ALDEx2.1']</span></span>
<span><span class="co"># DA_methods &lt;- DA_methods[!names(DA_methods) == 'DA_corncob.1']</span></span>
<span><span class="va">DA_methods</span> <span class="op">&lt;-</span> <span class="va">DA_methods</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span> <span class="op">==</span> <span class="st">'DA_edgeR.1'</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "DA_DESeq2.1"        "DA_DESeq2.1"        "DA_limma.1"        </span></span>
<span><span class="co">##  [4] "DA_limma.1"         "DA_metagenomeSeq.1" "DA_ALDEx2.1"       </span></span>
<span><span class="co">##  [7] "DA_MAST.1"          "DA_Seurat.1"        "ancombc.1"         </span></span>
<span><span class="co">## [10] "wilcox.3"           "wilcox.4"           "ZINQ.9"            </span></span>
<span><span class="co">## [13] "ZINQ.10"            "lefse.12"           "lefse.13"</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="va">DA_output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">DA_output</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span></span>
<span>            <span class="st">"Running method "</span>, <span class="va">i</span>, <span class="st">": "</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">" - "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="op">)</span></span>
<span>        <span class="va">DA_output</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/conditions.html" class="external-link">tryCatch</a></span><span class="op">(</span></span>
<span>            error <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">e</span><span class="op">)</span> <span class="cn">NULL</span>,</span>
<span>            <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/runDA.html" class="external-link">runDA</a></span><span class="op">(</span><span class="va">DA_methods</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="va">ps</span>, weights <span class="op">=</span> <span class="va">zw</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="va">DA_output</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_flatten.html" class="external-link">list_flatten</a></span><span class="op">(</span><span class="va">DA_output</span>, name_spec <span class="op">=</span> <span class="st">"{inner}"</span><span class="op">)</span></span>
<span>    <span class="va">DA_output</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/keep.html" class="external-link">discard</a></span><span class="op">(</span><span class="va">DA_output</span>, <span class="va">is.null</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running method 1: DA_DESeq2.1 - 2024-05-29 20:04:57.477826</span></span></code></pre>
<pre><code><span><span class="co">## Running method 2: DA_DESeq2.1 - 2024-05-29 20:05:02.707853</span></span></code></pre>
<pre><code><span><span class="co">## Running method 3: DA_limma.1 - 2024-05-29 20:05:09.351876</span></span></code></pre>
<pre><code><span><span class="co">## Running method 4: DA_limma.1 - 2024-05-29 20:05:09.374655</span></span></code></pre>
<pre><code><span><span class="co">## Running method 5: DA_metagenomeSeq.1 - 2024-05-29 20:05:09.395756</span></span></code></pre>
<pre><code><span><span class="co">## Running method 6: DA_ALDEx2.1 - 2024-05-29 20:05:09.879802</span></span></code></pre>
<pre><code><span><span class="co">## Running method 7: DA_MAST.1 - 2024-05-29 20:05:11.716792</span></span></code></pre>
<pre><code><span><span class="co">## Running method 8: DA_Seurat.1 - 2024-05-29 20:05:13.027448</span></span></code></pre>
<pre><code><span><span class="co">## For a (much!) faster implementation of the Wilcoxon Rank Sum Test,</span></span>
<span><span class="co">## (default method for FindMarkers) please install the presto package</span></span>
<span><span class="co">## --------------------------------------------</span></span>
<span><span class="co">## install.packages('devtools')</span></span>
<span><span class="co">## devtools::install_github('immunogenomics/presto')</span></span>
<span><span class="co">## --------------------------------------------</span></span>
<span><span class="co">## After installation of presto, Seurat will automatically use the more </span></span>
<span><span class="co">## efficient implementation (no further action necessary).</span></span>
<span><span class="co">## This message will be shown once per session</span></span></code></pre>
<pre><code><span><span class="co">## Running method 9: ancombc.1 - 2024-05-29 20:05:13.543902</span></span></code></pre>
<pre><code><span><span class="co">## 'ancombc' has been fully evolved to 'ancombc2'. </span></span>
<span><span class="co">## Explore the enhanced capabilities of our refined method!</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: foreach</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'foreach'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:purrr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     accumulate, when</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: rngtools</span></span></code></pre>
<pre><code><span><span class="co">## Running method 10: wilcox.3 - 2024-05-29 20:05:37.472812</span></span></code></pre>
<pre><code><span><span class="co">## Running method 11: wilcox.4 - 2024-05-29 20:05:37.649764</span></span></code></pre>
<pre><code><span><span class="co">## Running method 12: ZINQ.9 - 2024-05-29 20:05:37.777002</span></span></code></pre>
<pre><code><span><span class="co">## Running method 13: ZINQ.10 - 2024-05-29 20:05:38.697752</span></span></code></pre>
<pre><code><span><span class="co">## Running method 14: lefse.12 - 2024-05-29 20:05:39.469065</span></span></code></pre>
<pre><code><span><span class="co">## Running method 15: lefse.13 - 2024-05-29 20:05:41.064052</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tim</span></span></code></pre></div>
<pre><code><span><span class="co">##    user  system elapsed </span></span>
<span><span class="co">##  46.850   8.134  45.628</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="enrichment">Enrichment<a class="anchor" aria-label="anchor" href="#enrichment"></a>
</h2>
<p>Get direction</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">direction</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_direction_cols.html">get_direction_cols</a></span><span class="op">(</span><span class="va">DA_output</span>, <span class="va">conditions_col</span>, <span class="va">conditions</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="enrichment-adjp-0-1">Enrichment (adjP &lt;= 0.1)<a class="anchor" aria-label="anchor" href="#enrichment-adjp-0-1"></a>
</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/createEnrichment.html" class="external-link">createEnrichment</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">DA_output</span>,</span>
<span>    priorKnowledge <span class="op">=</span> <span class="va">prior_info</span>,</span>
<span>    enrichmentCol <span class="op">=</span> <span class="st">"taxon_annotation"</span>,</span>
<span>    namesCol <span class="op">=</span> <span class="st">"taxon_name"</span>,</span>
<span>    slot <span class="op">=</span> <span class="st">"pValMat"</span>, colName <span class="op">=</span> <span class="st">"adjP"</span>, type <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>    direction <span class="op">=</span> <span class="va">direction</span>,</span>
<span>    threshold_pvalue <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    threshold_logfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    top <span class="op">=</span> <span class="cn">NULL</span>, <span class="co"># No top feature selected</span></span>
<span>    alternative <span class="op">=</span> <span class="st">"greater"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span> </span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-enrichment">Plot enrichment<a class="anchor" aria-label="anchor" href="#plot-enrichment"></a>
</h3>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrich_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>    enrichment <span class="op">=</span> <span class="va">enrichment</span>, </span>
<span>    enrichment_col <span class="op">=</span> <span class="st">"taxon_annotation"</span>,</span>
<span>    levels_to_plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aerobic"</span>, <span class="st">"anaerobic"</span>, <span class="st">"facultative_anaerobic"</span><span class="op">)</span>,</span>
<span>    conditions <span class="op">=</span> <span class="va">conditions</span></span>
<span><span class="op">)</span></span>
<span><span class="va">enrich_plot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_enrichment_2.html">plot_enrichment_2</a></span><span class="op">(</span></span>
<span>    <span class="va">enrich_plot</span>,</span>
<span>    dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>up <span class="op">=</span> <span class="st">'Sup Plq'</span>, down <span class="op">=</span> <span class="st">'Sub Plq'</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">enrich_plot2</span></span></code></pre></div>
<p><img src="HMP_2012_WMS_gingival_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="putative-true-positives---putative-false-positives">Putative true positives - putative false positives<a class="anchor" aria-label="anchor" href="#putative-true-positives---putative-false-positives"></a>
</h2>
<div class="section level3">
<h3 id="calculate-tp---fp-ratio-no-threshold">Calculate TP - FP ratio (no threshold)<a class="anchor" aria-label="anchor" href="#calculate-tp---fp-ratio-no-threshold"></a>
</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">positives</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/createPositives.html" class="external-link">createPositives</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">DA_output</span>, </span>
<span>    priorKnowledge <span class="op">=</span> <span class="va">prior_info</span>, </span>
<span>    enrichmentCol <span class="op">=</span> <span class="st">"taxon_annotation"</span>, namesCol <span class="op">=</span> <span class="st">"taxon_name"</span>,</span>
<span>    slot <span class="op">=</span> <span class="st">"pValMat"</span>, colName <span class="op">=</span> <span class="st">"adjP"</span>, type <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>    direction <span class="op">=</span> <span class="va">direction</span>,</span>
<span>    threshold_pvalue <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    threshold_logfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    top <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq.int</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fl">0</span>, to <span class="op">=</span> <span class="fl">50</span>, by <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,</span>
<span>    alternative <span class="op">=</span> <span class="st">"greater"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    TP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DOWN Abundant"</span>, <span class="st">"anaerobic"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UP Abundant"</span>, <span class="st">"aerobic"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    FP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DOWN Abundant"</span>, <span class="st">"aerobic"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UP Abundant"</span>, <span class="st">"anaerobic"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/mia/man/deprecate.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_meth_class.html">get_meth_class</a></span><span class="op">(</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'method'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-tp---fp">Plot TP - FP<a class="anchor" aria-label="anchor" href="#plot-tp---fp"></a>
</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">positive_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_positives.html">plot_positives</a></span><span class="op">(</span><span class="va">positives</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span> <span class="op">~</span> <span class="op">{</span></span>
<span>        <span class="va">.x</span> <span class="op">+</span></span>
<span>            <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>                axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">17</span><span class="op">)</span>,</span>
<span>                axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,</span>
<span>                legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span>,</span>
<span>                strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">17</span><span class="op">)</span></span>
<span>            <span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">positive_plots</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="HMP_2012_WMS_gingival_files/figure-html/unnamed-chunk-8-1.png" width="1440"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] doRNG_1.8.6                           </span></span>
<span><span class="co">##  [2] rngtools_1.5.2                        </span></span>
<span><span class="co">##  [3] foreach_1.5.2                         </span></span>
<span><span class="co">##  [4] gridExtra_2.3                         </span></span>
<span><span class="co">##  [5] ggplot2_3.5.1                         </span></span>
<span><span class="co">##  [6] purrr_1.0.2                           </span></span>
<span><span class="co">##  [7] dplyr_1.1.4                           </span></span>
<span><span class="co">##  [8] benchdamic_1.9.4                      </span></span>
<span><span class="co">##  [9] phyloseq_1.48.0                       </span></span>
<span><span class="co">## [10] mia_1.12.0                            </span></span>
<span><span class="co">## [11] MultiAssayExperiment_1.30.1           </span></span>
<span><span class="co">## [12] MicrobiomeBenchmarkData_1.6.0         </span></span>
<span><span class="co">## [13] TreeSummarizedExperiment_2.12.0       </span></span>
<span><span class="co">## [14] Biostrings_2.72.0                     </span></span>
<span><span class="co">## [15] XVector_0.44.0                        </span></span>
<span><span class="co">## [16] SingleCellExperiment_1.26.0           </span></span>
<span><span class="co">## [17] SummarizedExperiment_1.34.0           </span></span>
<span><span class="co">## [18] Biobase_2.64.0                        </span></span>
<span><span class="co">## [19] GenomicRanges_1.56.0                  </span></span>
<span><span class="co">## [20] GenomeInfoDb_1.40.1                   </span></span>
<span><span class="co">## [21] IRanges_2.38.0                        </span></span>
<span><span class="co">## [22] S4Vectors_0.42.0                      </span></span>
<span><span class="co">## [23] BiocGenerics_0.50.0                   </span></span>
<span><span class="co">## [24] MatrixGenerics_1.16.0                 </span></span>
<span><span class="co">## [25] matrixStats_1.3.0                     </span></span>
<span><span class="co">## [26] MicrobiomeBenchmarkDataAnalyses_0.99.8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] igraph_2.0.3                ica_1.0-3                  </span></span>
<span><span class="co">##   [3] plotly_4.10.4               Formula_1.2-5              </span></span>
<span><span class="co">##   [5] scater_1.32.0               zlibbioc_1.50.0            </span></span>
<span><span class="co">##   [7] tidyselect_1.2.1            bit_4.0.5                  </span></span>
<span><span class="co">##   [9] doParallel_1.0.17           clue_0.3-65                </span></span>
<span><span class="co">##  [11] lattice_0.22-6              blob_1.2.4                 </span></span>
<span><span class="co">##  [13] stringr_1.5.1               S4Arrays_1.4.1             </span></span>
<span><span class="co">##  [15] tidytext_0.4.2              parallel_4.4.0             </span></span>
<span><span class="co">##  [17] png_0.1-8                   ZINQ_1.0                   </span></span>
<span><span class="co">##  [19] cli_3.6.2                   CVXR_1.0-12                </span></span>
<span><span class="co">##  [21] multtest_2.60.0             goftest_1.2-3              </span></span>
<span><span class="co">##  [23] pkgdown_2.0.9               textshaping_0.4.0          </span></span>
<span><span class="co">##  [25] bluster_1.14.0              tokenizers_0.3.0           </span></span>
<span><span class="co">##  [27] BiocNeighbors_1.22.0        uwot_0.2.2                 </span></span>
<span><span class="co">##  [29] curl_5.2.1                  zinbwave_1.26.0            </span></span>
<span><span class="co">##  [31] mime_0.12                   evaluate_0.23              </span></span>
<span><span class="co">##  [33] tidytree_0.4.6              leiden_0.4.3.1             </span></span>
<span><span class="co">##  [35] coin_1.4-3                  stringi_1.8.4              </span></span>
<span><span class="co">##  [37] corncob_0.4.1               backports_1.5.0            </span></span>
<span><span class="co">##  [39] fBasics_4032.96             desc_1.4.3                 </span></span>
<span><span class="co">##  [41] XML_3.99-0.16.1             lmerTest_3.1-3             </span></span>
<span><span class="co">##  [43] gsl_2.1-8                   Exact_3.2                  </span></span>
<span><span class="co">##  [45] httpuv_1.6.15               Wrench_1.22.0              </span></span>
<span><span class="co">##  [47] AnnotationDbi_1.66.0        magrittr_2.0.3             </span></span>
<span><span class="co">##  [49] splines_4.4.0               getopt_1.20.4              </span></span>
<span><span class="co">##  [51] jpeg_0.1-10                 pcaPP_2.0-4                </span></span>
<span><span class="co">##  [53] sctransform_0.4.1           rootSolve_1.8.2.4          </span></span>
<span><span class="co">##  [55] ggbeeswarm_0.7.2            statip_0.2.3               </span></span>
<span><span class="co">##  [57] DBI_1.2.2                   genefilter_1.86.0          </span></span>
<span><span class="co">##  [59] corpcor_1.6.10              jquerylib_0.1.4            </span></span>
<span><span class="co">##  [61] withr_3.0.0                 class_7.3-22               </span></span>
<span><span class="co">##  [63] systemfonts_1.1.0           lmtest_0.9-40              </span></span>
<span><span class="co">##  [65] mixOmics_6.28.0             htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [67] fs_1.6.4                    directlabels_2024.1.21     </span></span>
<span><span class="co">##  [69] ggrepel_0.9.5               labeling_0.4.3             </span></span>
<span><span class="co">##  [71] dearseq_1.16.0              SparseArray_1.4.8          </span></span>
<span><span class="co">##  [73] DESeq2_1.44.0               cellranger_1.1.0           </span></span>
<span><span class="co">##  [75] DEoptimR_1.1-3              spatial_7.3-17             </span></span>
<span><span class="co">##  [77] RcppZiggurat_0.1.6          annotate_1.82.0            </span></span>
<span><span class="co">##  [79] lmom_3.0                    truncnorm_1.0-9            </span></span>
<span><span class="co">##  [81] reticulate_1.37.0           zoo_1.8-12                 </span></span>
<span><span class="co">##  [83] knitr_1.46                  UCSC.utils_1.0.0           </span></span>
<span><span class="co">##  [85] modeest_2.4.0               decontam_1.24.0            </span></span>
<span><span class="co">##  [87] fansi_1.0.6                 patchwork_1.2.0            </span></span>
<span><span class="co">##  [89] caTools_1.18.2              grid_4.4.0                 </span></span>
<span><span class="co">##  [91] data.table_1.15.4           timeDate_4032.109          </span></span>
<span><span class="co">##  [93] biglm_0.9-2.1               rhdf5_2.48.0               </span></span>
<span><span class="co">##  [95] vegan_2.6-6.1               quantreg_5.98              </span></span>
<span><span class="co">##  [97] RSpectra_0.16-1             irlba_2.3.5.1              </span></span>
<span><span class="co">##  [99] janeaustenr_1.0.0           fastDummies_1.7.3          </span></span>
<span><span class="co">## [101] DescTools_0.99.54           lazyeval_0.2.2             </span></span>
<span><span class="co">## [103] yaml_2.3.8                  ade4_1.7-22                </span></span>
<span><span class="co">## [105] survival_3.6-4              scattermore_1.2            </span></span>
<span><span class="co">## [107] crayon_1.5.2                RcppAnnoy_0.0.22           </span></span>
<span><span class="co">## [109] RColorBrewer_1.1-3          tidyr_1.3.1                </span></span>
<span><span class="co">## [111] progressr_0.14.0            later_1.3.2                </span></span>
<span><span class="co">## [113] zCompositions_1.5.0-3       ggridges_0.5.6             </span></span>
<span><span class="co">## [115] codetools_0.2-20            base64enc_0.1-3            </span></span>
<span><span class="co">## [117] KEGGREST_1.44.0             Seurat_5.1.0               </span></span>
<span><span class="co">## [119] Rtsne_0.17                  shape_1.4.6.1              </span></span>
<span><span class="co">## [121] limma_3.60.2                filelock_1.0.3             </span></span>
<span><span class="co">## [123] foreign_0.8-86              pkgconfig_2.0.3            </span></span>
<span><span class="co">## [125] MicrobiomeStat_1.2          rmutil_1.1.10              </span></span>
<span><span class="co">## [127] spatstat.sparse_3.0-3       ape_5.8                    </span></span>
<span><span class="co">## [129] viridisLite_0.4.2           xtable_1.8-4               </span></span>
<span><span class="co">## [131] interp_1.1-6                highr_0.11                 </span></span>
<span><span class="co">## [133] plyr_1.8.9                  httr_1.4.7                 </span></span>
<span><span class="co">## [135] rbibutils_2.2.16            tools_4.4.0                </span></span>
<span><span class="co">## [137] globals_0.16.3              SeuratObject_5.0.2         </span></span>
<span><span class="co">## [139] Rfast_2.1.0                 beeswarm_0.4.0             </span></span>
<span><span class="co">## [141] htmlTable_2.4.2             checkmate_2.3.1            </span></span>
<span><span class="co">## [143] nlme_3.1-164                dbplyr_2.5.0               </span></span>
<span><span class="co">## [145] MatrixModels_0.5-3          lme4_1.1-35.3              </span></span>
<span><span class="co">## [147] digest_0.6.35               optparse_1.7.5             </span></span>
<span><span class="co">## [149] permute_0.9-7               numDeriv_2016.8-1.1        </span></span>
<span><span class="co">## [151] Matrix_1.7-0                farver_2.1.2               </span></span>
<span><span class="co">## [153] reshape2_1.4.4              SnowballC_0.7.1            </span></span>
<span><span class="co">## [155] yulab.utils_0.1.4           viridis_0.6.5              </span></span>
<span><span class="co">## [157] DirichletMultinomial_1.46.0 rpart_4.1.23               </span></span>
<span><span class="co">## [159] glue_1.7.0                  cachem_1.1.0               </span></span>
<span><span class="co">## [161] BiocFileCache_2.12.0        polyclip_1.10-6            </span></span>
<span><span class="co">## [163] Hmisc_5.1-3                 generics_0.1.3             </span></span>
<span><span class="co">## [165] CompQuadForm_1.4.3          mvtnorm_1.2-5              </span></span>
<span><span class="co">## [167] NADA_1.6-1.1                metagenomeSeq_1.46.0       </span></span>
<span><span class="co">## [169] survey_4.4-2                parallelly_1.37.1          </span></span>
<span><span class="co">## [171] statmod_1.5.0               biomformat_1.32.0          </span></span>
<span><span class="co">## [173] RcppHNSW_0.6.0              ragg_1.3.2                 </span></span>
<span><span class="co">## [175] ScaledMatrix_1.12.0         minqa_1.2.7                </span></span>
<span><span class="co">## [177] pbapply_1.7-2               ANCOMBC_2.6.0              </span></span>
<span><span class="co">## [179] timeSeries_4032.109         glmnet_4.1-8               </span></span>
<span><span class="co">## [181] spam_2.10-0                 utf8_1.2.4                 </span></span>
<span><span class="co">## [183] ALDEx2_1.36.0               mitools_2.4                </span></span>
<span><span class="co">## [185] gtools_3.9.5                readxl_1.4.3               </span></span>
<span><span class="co">## [187] softImpute_1.4-1            shiny_1.8.1.1              </span></span>
<span><span class="co">## [189] GenomeInfoDbData_1.2.12     energy_1.7-11              </span></span>
<span><span class="co">## [191] rhdf5filters_1.16.0         memoise_2.0.1              </span></span>
<span><span class="co">## [193] rmarkdown_2.27              scales_1.3.0               </span></span>
<span><span class="co">## [195] stabledist_0.7-1            gld_2.6.6                  </span></span>
<span><span class="co">## [197] future_1.33.2               RANN_2.6.1                 </span></span>
<span><span class="co">## [199] spatstat.data_3.0-4         rstudioapi_0.16.0          </span></span>
<span><span class="co">## [201] cluster_2.1.6               Maaslin2_1.18.0            </span></span>
<span><span class="co">## [203] hms_1.1.3                   GUniFrac_1.8               </span></span>
<span><span class="co">## [205] spatstat.utils_3.0-4        fitdistrplus_1.1-11        </span></span>
<span><span class="co">## [207] munsell_0.5.1               cowplot_1.1.3              </span></span>
<span><span class="co">## [209] ellipse_0.5.0               colorspace_2.1-0           </span></span>
<span><span class="co">## [211] rlang_1.1.3                 ggdendro_0.2.0             </span></span>
<span><span class="co">## [213] quadprog_1.5-8              DelayedMatrixStats_1.26.0  </span></span>
<span><span class="co">## [215] sparseMatrixStats_1.16.0    dotCall64_1.1-1            </span></span>
<span><span class="co">## [217] scuttle_1.14.0              mgcv_1.9-1                 </span></span>
<span><span class="co">## [219] xfun_0.44                   NOISeq_2.48.0              </span></span>
<span><span class="co">## [221] e1071_1.7-14                TH.data_1.1-2              </span></span>
<span><span class="co">## [223] rARPACK_0.11-0              iterators_1.0.14           </span></span>
<span><span class="co">## [225] modeltools_0.2-23           abind_1.4-5                </span></span>
<span><span class="co">## [227] tibble_3.2.1                libcoin_1.0-10             </span></span>
<span><span class="co">## [229] treeio_1.28.0               gmp_0.7-4                  </span></span>
<span><span class="co">## [231] Rhdf5lib_1.26.0             DECIPHER_3.0.0             </span></span>
<span><span class="co">## [233] bitops_1.0-7                Rdpack_2.6                 </span></span>
<span><span class="co">## [235] promises_1.3.0              inline_0.3.19              </span></span>
<span><span class="co">## [237] RSQLite_2.3.7               sandwich_3.1-0             </span></span>
<span><span class="co">## [239] DelayedArray_0.30.1         proxy_0.4-27               </span></span>
<span><span class="co">## [241] Rmpfr_0.9-5                 compiler_4.4.0             </span></span>
<span><span class="co">## [243] forcats_1.0.0               prettyunits_1.2.0          </span></span>
<span><span class="co">## [245] boot_1.3-30                 beachmat_2.20.0            </span></span>
<span><span class="co">## [247] SparseM_1.82                microbiome_1.26.0          </span></span>
<span><span class="co">## [249] listenv_0.9.1               Rcpp_1.0.12                </span></span>
<span><span class="co">## [251] edgeR_4.2.0                 BiocSingular_1.20.0        </span></span>
<span><span class="co">## [253] tensor_1.5                  progress_1.2.3             </span></span>
<span><span class="co">## [255] MASS_7.3-60.2               BiocParallel_1.38.0        </span></span>
<span><span class="co">## [257] MAST_1.30.0                 stable_1.1.6               </span></span>
<span><span class="co">## [259] spatstat.random_3.2-3       R6_2.5.1                   </span></span>
<span><span class="co">## [261] fastmap_1.2.0               multcomp_1.4-25            </span></span>
<span><span class="co">## [263] vipor_0.4.7                 ROCR_1.0-11                </span></span>
<span><span class="co">## [265] rsvd_1.0.5                  nnet_7.3-19                </span></span>
<span><span class="co">## [267] gtable_0.3.5                KernSmooth_2.23-22         </span></span>
<span><span class="co">## [269] latticeExtra_0.6-30         miniUI_0.1.1.1             </span></span>
<span><span class="co">## [271] deldir_2.0-4                htmltools_0.5.8.1          </span></span>
<span><span class="co">## [273] RcppParallel_5.1.7          bit64_4.0.5                </span></span>
<span><span class="co">## [275] spatstat.explore_3.2-7      lifecycle_1.0.4            </span></span>
<span><span class="co">## [277] MGLM_0.2.1                  nloptr_2.0.3               </span></span>
<span><span class="co">## [279] sass_0.4.9                  vctrs_0.6.5                </span></span>
<span><span class="co">## [281] robustbase_0.99-2           spatstat.geom_3.2-9        </span></span>
<span><span class="co">## [283] sp_2.1-4                    future.apply_1.11.2        </span></span>
<span><span class="co">## [285] bslib_0.7.0                 pillar_1.9.0               </span></span>
<span><span class="co">## [287] gplots_3.1.3.1              locfit_1.5-9.9             </span></span>
<span><span class="co">## [289] jsonlite_1.8.8              expm_0.999-9</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Samuel David Gamboa-Tuz.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
