<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MicrobiomeBenchmarkDataAnalyses">
<title>HMP_2012_16S_gingival_V35_subset - subgingival vs supragingival â€¢ MicrobiomeBenchmarkDataAnalyses</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HMP_2012_16S_gingival_V35_subset - subgingival vs supragingival">
<meta property="og:description" content="MicrobiomeBenchmarkDataAnalyses">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MicrobiomeBenchmarkDataAnalyses</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/MicrobiomeBenchmarkDataAnalyses.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/HMP_2012_16S_gingival_V13.html">HMP_2012_16S_gingival_V13 - subgingival vs supgragingival plaque</a>
    <a class="dropdown-item" href="../articles/HMP_2012_16S_gingival_V35.html">HMP_2012_16S_gingival_V35 - supgragingival vs sugingival plaque</a>
    <a class="dropdown-item" href="../articles/HMP_2012_16S_gingival_V35_subset.html">HMP_2012_16S_gingival_V35_subset - subgingival vs supragingival</a>
    <a class="dropdown-item" href="../articles/HMP_2012_WMS_gingival.html">HMP_2012_WMS_gingiva - Supragingival vs subgingival plaque</a>
    <a class="dropdown-item" href="../articles/Ravel_2011_16S_BV.html">Ravel_2011_16S_BV - Bacterial vaginosis</a>
    <a class="dropdown-item" href="../articles/Stammler_2016_16S_spikein.html">Sammler_2016_16S_spikein - Spike-in bacteria</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>HMP_2012_16S_gingival_V35_subset - subgingival vs supragingival</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses/blob/HEAD/vignettes/articles/HMP_2012_16S_gingival_V35_subset.Rmd" class="external-link"><code>vignettes/articles/HMP_2012_16S_gingival_V35_subset.Rmd</code></a></small>
      <div class="d-none name"><code>HMP_2012_16S_gingival_V35_subset.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/MicrobiomeBenchmarkDataAnalyses" class="external-link">MicrobiomeBenchmarkDataAnalyses</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/MicrobiomeBenchmarkData" class="external-link">MicrobiomeBenchmarkData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">benchdamic</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="import-data">Import data<a class="anchor" aria-label="anchor" href="#import-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_name</span> <span class="op">&lt;-</span><span class="st">'HMP_2012_16S_gingival_V35_subset'</span></span>
<span><span class="va">conditions_col</span> <span class="op">&lt;-</span> <span class="st">'body_subsite'</span></span>
<span><span class="va">conditions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>condB <span class="op">=</span> <span class="st">'subgingival_plaque'</span>, condA <span class="op">=</span> <span class="st">'supragingival_plaque'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MicrobiomeBenchmarkData/reference/getBenchmarkData.html" class="external-link">getBenchmarkData</a></span><span class="op">(</span><span class="va">dat_name</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterTaxa.html">filterTaxa</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="va">conditions_col</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="va">conditions_col</span><span class="op">]</span><span class="op">]</span>, levels <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 783 76 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(783): OTU_97.31247 OTU_97.44487 ... OTU_97.45365 OTU_97.45307</span></span>
<span><span class="co">## rowData names(7): kingdom phylum ... genus taxon_annotation</span></span>
<span><span class="co">## colnames(76): 700023057 700023179 ... 700114009 700114338</span></span>
<span><span class="co">## colData names(13): dataset subject_id ... sequencing_method</span></span>
<span><span class="co">##   variable_region_16s</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (783 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (892 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="prior-knowledge">Prior knowledge<a class="anchor" aria-label="anchor" href="#prior-knowledge"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">row_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">prior_info</span> <span class="op">&lt;-</span> <span class="va">row_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'genus'</span>, <span class="st">'taxon_annotation'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">prior_info</span><span class="op">$</span><span class="va">taxon_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">)</span></span>
<span><span class="va">prior_info</span><span class="op">$</span><span class="va">new_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">$</span><span class="va">taxon_name</span>, <span class="st">'|'</span>, <span class="va">prior_info</span><span class="op">$</span><span class="va">genus</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prior_info</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                      genus      taxon_annotation   taxon_name</span></span>
<span><span class="co">## OTU_97.31247 Streptococcus facultative_anaerobic OTU_97.31247</span></span>
<span><span class="co">## OTU_97.44487 Streptococcus facultative_anaerobic OTU_97.44487</span></span>
<span><span class="co">## OTU_97.34979 Streptococcus facultative_anaerobic OTU_97.34979</span></span>
<span><span class="co">## OTU_97.34572 Streptococcus facultative_anaerobic OTU_97.34572</span></span>
<span><span class="co">## OTU_97.42259 Streptococcus facultative_anaerobic OTU_97.42259</span></span>
<span><span class="co">## OTU_97.34698 Streptococcus facultative_anaerobic OTU_97.34698</span></span>
<span><span class="co">##                               new_names</span></span>
<span><span class="co">## OTU_97.31247 OTU_97.31247|Streptococcus</span></span>
<span><span class="co">## OTU_97.44487 OTU_97.44487|Streptococcus</span></span>
<span><span class="co">## OTU_97.34979 OTU_97.34979|Streptococcus</span></span>
<span><span class="co">## OTU_97.34572 OTU_97.34572|Streptococcus</span></span>
<span><span class="co">## OTU_97.42259 OTU_97.42259|Streptococcus</span></span>
<span><span class="co">## OTU_97.34698 OTU_97.34698|Streptococcus</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="convert-to-phyloseq">Convert to phyloseq<a class="anchor" aria-label="anchor" href="#convert-to-phyloseq"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/makePhyloseqFromTreeSummarizedExperiment.html" class="external-link">makePhyloseqFromTreeSummarizedExperiment</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">body_subsite</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">body_subsite</span><span class="op">)</span></span>
<span><span class="va">ps</span></span></code></pre></div>
<pre><code><span><span class="co">## phyloseq-class experiment-level object</span></span>
<span><span class="co">## otu_table()   OTU Table:         [ 783 taxa and 76 samples ]</span></span>
<span><span class="co">## sample_data() Sample Data:       [ 76 samples by 13 sample variables ]</span></span>
<span><span class="co">## tax_table()   Taxonomy Table:    [ 783 taxa by 6 taxonomic ranks ]</span></span>
<span><span class="co">## phy_tree()    Phylogenetic Tree: [ 783 tips and 778 internal nodes ]</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="run-da">Run DA<a class="anchor" aria-label="anchor" href="#run-da"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">time_start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">DA_output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DA.html">run_DA</a></span><span class="op">(</span><span class="va">ps</span>, <span class="va">conditions_col</span>, <span class="va">conditions</span><span class="op">)</span></span>
<span><span class="va">time_end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">time_elapsed</span> <span class="op">&lt;-</span> <span class="va">time_end</span> <span class="op">-</span> <span class="va">time_start</span></span>
<span><span class="va">time_elapsed</span></span></code></pre></div>
<pre><code><span><span class="co">## Time difference of 3.165193 mins</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="enrichment">Enrichment<a class="anchor" aria-label="anchor" href="#enrichment"></a>
</h2>
<p>Get direction</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">direction</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_direction_cols.html">get_direction_cols</a></span><span class="op">(</span><span class="va">DA_output</span>, <span class="va">conditions_col</span>, <span class="va">conditions</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="enrichment-adjp-0-1">Enrichment (adjP &lt;= 0.1)<a class="anchor" aria-label="anchor" href="#enrichment-adjp-0-1"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/createEnrichment.html" class="external-link">createEnrichment</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">DA_output</span>,</span>
<span>    priorKnowledge <span class="op">=</span> <span class="va">prior_info</span>,</span>
<span>    enrichmentCol <span class="op">=</span> <span class="st">"taxon_annotation"</span>,</span>
<span>    namesCol <span class="op">=</span> <span class="st">"new_names"</span>,</span>
<span>    slot <span class="op">=</span> <span class="st">"pValMat"</span>, colName <span class="op">=</span> <span class="st">"adjP"</span>, type <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>    direction <span class="op">=</span> <span class="va">direction</span>,</span>
<span>    threshold_pvalue <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    threshold_logfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    top <span class="op">=</span> <span class="cn">NULL</span>, <span class="co"># No top feature selected</span></span>
<span>    alternative <span class="op">=</span> <span class="st">"greater"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span> </span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-enrichment">Plot enrichment<a class="anchor" aria-label="anchor" href="#plot-enrichment"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrich_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>    enrichment <span class="op">=</span> <span class="va">enrichment</span>, </span>
<span>    enrichment_col <span class="op">=</span> <span class="st">"taxon_annotation"</span>,</span>
<span>    levels_to_plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aerobic"</span>, <span class="st">"anaerobic"</span>, <span class="st">"facultative_anaerobic"</span><span class="op">)</span>,</span>
<span>    conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>condB <span class="op">=</span> <span class="st">'subgingival'</span>, condA <span class="op">=</span> <span class="st">'supragingival'</span><span class="op">)</span> </span>
<span><span class="op">)</span></span>
<span><span class="va">enrich_plot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_enrichment_2.html">plot_enrichment_2</a></span><span class="op">(</span></span>
<span>    <span class="va">enrich_plot</span>,</span>
<span>    dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>up <span class="op">=</span> <span class="st">'Suptragingival Plaque'</span>, down <span class="op">=</span> <span class="st">'Subgingival Plaque'</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">enrich_plot2</span></span></code></pre></div>
<p><img src="HMP_2012_16S_gingival_V35_subset_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="putative-true-positives---putative-false-positives">Putative true positives - putative false positives<a class="anchor" aria-label="anchor" href="#putative-true-positives---putative-false-positives"></a>
</h2>
<div class="section level3">
<h3 id="calculate-tp---fp-ratio-no-threshold">Calculate TP - FP ratio (no threshold)<a class="anchor" aria-label="anchor" href="#calculate-tp---fp-ratio-no-threshold"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">positives</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/benchdamic/man/createPositives.html" class="external-link">createPositives</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">DA_output</span>, </span>
<span>    priorKnowledge <span class="op">=</span> <span class="va">prior_info</span>, </span>
<span>    enrichmentCol <span class="op">=</span> <span class="st">"taxon_annotation"</span>, namesCol <span class="op">=</span> <span class="st">"new_names"</span>,</span>
<span>    slot <span class="op">=</span> <span class="st">"pValMat"</span>, colName <span class="op">=</span> <span class="st">"adjP"</span>, type <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>    direction <span class="op">=</span> <span class="va">direction</span>,</span>
<span>    threshold_pvalue <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    threshold_logfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    top <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq.int</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fl">0</span>, to <span class="op">=</span> <span class="fl">50</span>, by <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,</span>
<span>    alternative <span class="op">=</span> <span class="st">"greater"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    TP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DOWN Abundant"</span>, <span class="st">"anaerobic"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UP Abundant"</span>, <span class="st">"aerobic"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    FP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DOWN Abundant"</span>, <span class="st">"aerobic"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UP Abundant"</span>, <span class="st">"anaerobic"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_meth_class.html">get_meth_class</a></span><span class="op">(</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'method'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-tp---fp">Plot TP - FP<a class="anchor" aria-label="anchor" href="#plot-tp---fp"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">positive_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_positives.html">plot_positives</a></span><span class="op">(</span><span class="va">positives</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">positive_plots</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="HMP_2012_16S_gingival_V35_subset_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] gridExtra_2.3                             </span></span>
<span><span class="co">##  [2] ggplot2_3.3.6                             </span></span>
<span><span class="co">##  [3] dplyr_1.0.10                              </span></span>
<span><span class="co">##  [4] benchdamic_1.2.5                          </span></span>
<span><span class="co">##  [5] mia_1.4.0                                 </span></span>
<span><span class="co">##  [6] MultiAssayExperiment_1.22.0               </span></span>
<span><span class="co">##  [7] MicrobiomeBenchmarkData_0.99.0            </span></span>
<span><span class="co">##  [8] TreeSummarizedExperiment_2.4.0            </span></span>
<span><span class="co">##  [9] Biostrings_2.64.1                         </span></span>
<span><span class="co">## [10] XVector_0.36.0                            </span></span>
<span><span class="co">## [11] SingleCellExperiment_1.18.0               </span></span>
<span><span class="co">## [12] SummarizedExperiment_1.26.1               </span></span>
<span><span class="co">## [13] Biobase_2.56.0                            </span></span>
<span><span class="co">## [14] GenomicRanges_1.48.0                      </span></span>
<span><span class="co">## [15] GenomeInfoDb_1.32.4                       </span></span>
<span><span class="co">## [16] IRanges_2.30.1                            </span></span>
<span><span class="co">## [17] S4Vectors_0.34.0                          </span></span>
<span><span class="co">## [18] BiocGenerics_0.42.0                       </span></span>
<span><span class="co">## [19] MatrixGenerics_1.8.1                      </span></span>
<span><span class="co">## [20] matrixStats_0.62.0                        </span></span>
<span><span class="co">## [21] MicrobiomeBenchmarkDataAnalyses_0.0.0.9000</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rsvd_1.0.5                  zinbwave_1.18.0            </span></span>
<span><span class="co">##   [3] Hmisc_4.7-1                 ica_1.0-3                  </span></span>
<span><span class="co">##   [5] ZINQ_1.0                    class_7.3-20               </span></span>
<span><span class="co">##   [7] glmnet_4.1-4                foreach_1.5.2              </span></span>
<span><span class="co">##   [9] lmtest_0.9-40               rprojroot_2.0.3            </span></span>
<span><span class="co">##  [11] crayon_1.5.1                rbibutils_2.2.9            </span></span>
<span><span class="co">##  [13] spatstat.core_2.4-4         MASS_7.3-58.1              </span></span>
<span><span class="co">##  [15] rhdf5filters_1.8.0          MAST_1.22.0                </span></span>
<span><span class="co">##  [17] nlme_3.1-159                backports_1.4.1            </span></span>
<span><span class="co">##  [19] ALDEx2_1.28.1               rlang_1.0.6                </span></span>
<span><span class="co">##  [21] ROCR_1.0-11                 readxl_1.4.1               </span></span>
<span><span class="co">##  [23] SparseM_1.81                microbiome_1.18.0          </span></span>
<span><span class="co">##  [25] irlba_2.3.5                 nloptr_2.0.3               </span></span>
<span><span class="co">##  [27] limma_3.52.3                scater_1.24.0              </span></span>
<span><span class="co">##  [29] filelock_1.0.2              BiocParallel_1.30.3        </span></span>
<span><span class="co">##  [31] bit64_4.0.5                 glue_1.6.2                 </span></span>
<span><span class="co">##  [33] rngtools_1.5.2              sctransform_0.3.5          </span></span>
<span><span class="co">##  [35] parallel_4.2.1              vipor_0.4.5                </span></span>
<span><span class="co">##  [37] spatstat.sparse_2.1-1       AnnotationDbi_1.58.0       </span></span>
<span><span class="co">##  [39] spatstat.geom_2.4-0         tidyselect_1.1.2           </span></span>
<span><span class="co">##  [41] SeuratObject_4.1.2          NADA_1.6-1.1               </span></span>
<span><span class="co">##  [43] fitdistrplus_1.1-8          phyloseq_1.40.0            </span></span>
<span><span class="co">##  [45] XML_3.99-0.10               tidyr_1.2.1                </span></span>
<span><span class="co">##  [47] zoo_1.8-11                  MatrixModels_0.5-1         </span></span>
<span><span class="co">##  [49] xtable_1.8-4                magrittr_2.0.3             </span></span>
<span><span class="co">##  [51] evaluate_0.16               Rdpack_2.4                 </span></span>
<span><span class="co">##  [53] scuttle_1.6.3               cli_3.4.1                  </span></span>
<span><span class="co">##  [55] zlibbioc_1.42.0             rstudioapi_0.14            </span></span>
<span><span class="co">##  [57] doRNG_1.8.2                 miniUI_0.1.1.1             </span></span>
<span><span class="co">##  [59] sp_1.5-0                    metagenomeSeq_1.38.0       </span></span>
<span><span class="co">##  [61] bslib_0.4.0                 rpart_4.1.16               </span></span>
<span><span class="co">##  [63] zCompositions_1.4.0-1       tidytext_0.3.4             </span></span>
<span><span class="co">##  [65] treeio_1.20.2               shiny_1.7.2                </span></span>
<span><span class="co">##  [67] BiocSingular_1.12.0         xfun_0.33                  </span></span>
<span><span class="co">##  [69] multtest_2.52.0             cluster_2.1.4              </span></span>
<span><span class="co">##  [71] caTools_1.18.2              biomformat_1.24.0          </span></span>
<span><span class="co">##  [73] KEGGREST_1.36.3             quantreg_5.94              </span></span>
<span><span class="co">##  [75] tibble_3.1.8                expm_0.999-6               </span></span>
<span><span class="co">##  [77] ggrepel_0.9.1               ape_5.6-2                  </span></span>
<span><span class="co">##  [79] listenv_0.8.0               png_0.1-7                  </span></span>
<span><span class="co">##  [81] permute_0.9-7               future_1.28.0              </span></span>
<span><span class="co">##  [83] withr_2.5.0                 bitops_1.0-7               </span></span>
<span><span class="co">##  [85] plyr_1.8.7                  cellranger_1.1.0           </span></span>
<span><span class="co">##  [87] RcppZiggurat_0.1.6          survey_4.1-1               </span></span>
<span><span class="co">##  [89] e1071_1.7-11                pillar_1.8.1               </span></span>
<span><span class="co">##  [91] gplots_3.1.3                cachem_1.0.6               </span></span>
<span><span class="co">##  [93] multcomp_1.4-20             fs_1.5.2                   </span></span>
<span><span class="co">##  [95] DelayedMatrixStats_1.18.0   vctrs_0.4.1                </span></span>
<span><span class="co">##  [97] ellipsis_0.3.2              generics_0.1.3             </span></span>
<span><span class="co">##  [99] tools_4.2.1                 foreign_0.8-82             </span></span>
<span><span class="co">## [101] NOISeq_2.40.0               beeswarm_0.4.0             </span></span>
<span><span class="co">## [103] munsell_0.5.0               proxy_0.4-27               </span></span>
<span><span class="co">## [105] DelayedArray_0.22.0         fastmap_1.1.0              </span></span>
<span><span class="co">## [107] compiler_4.2.1              abind_1.4-5                </span></span>
<span><span class="co">## [109] httpuv_1.6.6                DescTools_0.99.46          </span></span>
<span><span class="co">## [111] plotly_4.10.0               rgeos_0.5-9                </span></span>
<span><span class="co">## [113] decontam_1.16.0             GenomeInfoDbData_1.2.8     </span></span>
<span><span class="co">## [115] edgeR_3.38.4                lattice_0.20-45            </span></span>
<span><span class="co">## [117] deldir_1.0-6                utf8_1.2.2                 </span></span>
<span><span class="co">## [119] later_1.3.0                 BiocFileCache_2.4.0        </span></span>
<span><span class="co">## [121] jsonlite_1.8.0              tokenizers_0.2.3           </span></span>
<span><span class="co">## [123] scales_1.2.1                gld_2.6.5                  </span></span>
<span><span class="co">## [125] ScaledMatrix_1.4.1          tidytree_0.4.1             </span></span>
<span><span class="co">## [127] pbapply_1.5-0               sparseMatrixStats_1.8.0    </span></span>
<span><span class="co">## [129] genefilter_1.78.0           lazyeval_0.2.2             </span></span>
<span><span class="co">## [131] promises_1.2.0.1            doParallel_1.0.17          </span></span>
<span><span class="co">## [133] latticeExtra_0.6-30         goftest_1.2-3              </span></span>
<span><span class="co">## [135] spatstat.utils_2.3-1        reticulate_1.26            </span></span>
<span><span class="co">## [137] checkmate_2.1.0             rmarkdown_2.16             </span></span>
<span><span class="co">## [139] pkgdown_2.0.6               sandwich_3.0-2             </span></span>
<span><span class="co">## [141] cowplot_1.1.1               textshaping_0.3.6          </span></span>
<span><span class="co">## [143] statmod_1.4.37              forcats_0.5.2              </span></span>
<span><span class="co">## [145] Rtsne_0.16                  softImpute_1.4-1           </span></span>
<span><span class="co">## [147] uwot_0.1.14                 igraph_1.3.5               </span></span>
<span><span class="co">## [149] survival_3.4-0              yaml_2.3.5                 </span></span>
<span><span class="co">## [151] systemfonts_1.0.4           ANCOMBC_1.6.2              </span></span>
<span><span class="co">## [153] htmltools_0.5.3             memoise_2.0.1              </span></span>
<span><span class="co">## [155] modeltools_0.2-23           Seurat_4.2.0               </span></span>
<span><span class="co">## [157] locfit_1.5-9.6              viridisLite_0.4.1          </span></span>
<span><span class="co">## [159] digest_0.6.29               assertthat_0.2.1           </span></span>
<span><span class="co">## [161] mime_0.12                   rappdirs_0.3.3             </span></span>
<span><span class="co">## [163] RSQLite_2.2.17              Rfast_2.0.6                </span></span>
<span><span class="co">## [165] yulab.utils_0.0.5           future.apply_1.9.1         </span></span>
<span><span class="co">## [167] Exact_3.2                   data.table_1.14.2          </span></span>
<span><span class="co">## [169] blob_1.2.3                  vegan_2.6-2                </span></span>
<span><span class="co">## [171] ragg_1.2.2                  labeling_0.4.2             </span></span>
<span><span class="co">## [173] splines_4.2.1               Formula_1.2-4              </span></span>
<span><span class="co">## [175] DECIPHER_2.24.0             Rhdf5lib_1.18.2            </span></span>
<span><span class="co">## [177] RCurl_1.98-1.8              hms_1.1.2                  </span></span>
<span><span class="co">## [179] rhdf5_2.40.0                colorspace_2.0-3           </span></span>
<span><span class="co">## [181] base64enc_0.1-3             shape_1.4.6                </span></span>
<span><span class="co">## [183] ggbeeswarm_0.6.0            libcoin_1.0-9              </span></span>
<span><span class="co">## [185] nnet_7.3-17                 sass_0.4.2                 </span></span>
<span><span class="co">## [187] Rcpp_1.0.9                  coin_1.4-2                 </span></span>
<span><span class="co">## [189] RANN_2.6.1                  mvtnorm_1.1-3              </span></span>
<span><span class="co">## [191] fansi_1.0.3                 SnowballC_0.7.0            </span></span>
<span><span class="co">## [193] truncnorm_1.0-8             parallelly_1.32.1          </span></span>
<span><span class="co">## [195] R6_2.5.1                    grid_4.2.1                 </span></span>
<span><span class="co">## [197] ggridges_0.5.4              lifecycle_1.0.2            </span></span>
<span><span class="co">## [199] rootSolve_1.8.2.3           curl_4.3.2                 </span></span>
<span><span class="co">## [201] leiden_0.4.3                jquerylib_0.1.4            </span></span>
<span><span class="co">## [203] dearseq_1.8.4               Matrix_1.5-1               </span></span>
<span><span class="co">## [205] desc_1.4.2                  RcppAnnoy_0.0.19           </span></span>
<span><span class="co">## [207] TH.data_1.1-1               RColorBrewer_1.1-3         </span></span>
<span><span class="co">## [209] iterators_1.0.14            stringr_1.4.1              </span></span>
<span><span class="co">## [211] htmlwidgets_1.5.4           Wrench_1.14.0              </span></span>
<span><span class="co">## [213] polyclip_1.10-0             beachmat_2.12.0            </span></span>
<span><span class="co">## [215] purrr_0.3.4                 janeaustenr_1.0.0          </span></span>
<span><span class="co">## [217] mgcv_1.8-40                 globals_0.16.1             </span></span>
<span><span class="co">## [219] spatstat.random_2.2-0       lmom_2.9                   </span></span>
<span><span class="co">## [221] htmlTable_2.4.1             patchwork_1.1.2            </span></span>
<span><span class="co">## [223] progressr_0.11.0            codetools_0.2-18           </span></span>
<span><span class="co">## [225] prettyunits_1.1.1           gtools_3.9.3               </span></span>
<span><span class="co">## [227] dbplyr_2.2.1                gtable_0.3.1               </span></span>
<span><span class="co">## [229] DBI_1.1.3                   highr_0.9                  </span></span>
<span><span class="co">## [231] tensor_1.5                  httr_1.4.4                 </span></span>
<span><span class="co">## [233] KernSmooth_2.23-20          progress_1.2.2             </span></span>
<span><span class="co">## [235] stringi_1.7.8               farver_2.1.1               </span></span>
<span><span class="co">## [237] reshape2_1.4.4              annotate_1.74.0            </span></span>
<span><span class="co">## [239] viridis_0.6.2               ggdendro_0.1.23            </span></span>
<span><span class="co">## [241] boot_1.3-28                 BiocNeighbors_1.14.0       </span></span>
<span><span class="co">## [243] interp_1.1-3                ade4_1.7-19                </span></span>
<span><span class="co">## [245] geneplotter_1.74.0          CompQuadForm_1.4.3         </span></span>
<span><span class="co">## [247] energy_1.7-10               scattermore_0.8            </span></span>
<span><span class="co">## [249] DESeq2_1.36.0               bit_4.0.4                  </span></span>
<span><span class="co">## [251] jpeg_0.1-9                  spatstat.data_2.2-0        </span></span>
<span><span class="co">## [253] pkgconfig_2.0.3             gsl_2.1-7.1                </span></span>
<span><span class="co">## [255] DirichletMultinomial_1.38.0 MGLM_0.2.1                 </span></span>
<span><span class="co">## [257] mitools_2.4                 knitr_1.40</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Samuel David Gamboa-Tuz.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
